cmake/init-compilation-flags.cmake

```diff
@@ -49,7 +60,9 @@ endif()
 
 if(NOT DEFINED ENV{ENABLE_GRPROF})
```

Comment:
honestly all of the below options are almost useless, we can safely add `NOT APPLE` just in this condition

Reply:
let me leave as is, also I've added -dead_strip for macos - --gc-sections alternative.